netaddr.go

```diff
@@ -846,3 +871,422 @@ func (p *IPPrefix) UnmarshalText(text []byte) error {
 func (p IPPrefix) String() string {
 	return fmt.Sprintf("%s/%d", p.IP, p.Bits)
 }
+
+// LastIP returns the last IP in the prefix.
+func (p IPPrefix) LastIP() IP {
+	if p.IP.IsZero() {
+		return IP{}
+	}
+	a16 := p.IP.As16()
+	var off uint8
+	var bits uint8 = 128
+	if p.IP.Is4() {
+		off = 12
+		bits = 32
+	}
+	for b := p.Bits; b < bits; b++ {
+		byteNum, bitInByte := b/8, 7-(b%8)
+		a16[off+byteNum] |= 1 << uint(bitInByte)
+	}
+	if p.IP.Is4() {
+		return IPFrom16(a16)
+	} else {
+		return IPv6Raw(a16) // doesn't unmap
+	}
+}
+
+// IPRange represents an inclusive range of IP addresses
+// from the same address family.
+type IPRange struct {
+	// From is the starting IP address in the range.
+	// It must have the same address family (IPv4 vs IPv6) as To
```

Comment:
What happens if the contract is violated? AFAICT from below, we return a zero-ish value rather than, say, panic. Would be nice to document here.